pkg/kev/converter/kubernetes/project_service.go

```diff
@@ -320,24 +321,18 @@ func (p *ProjectService) serviceAccountName() string {
 	return p.SvcK8sConfig.Workload.ServiceAccountName
 }
 
-// restartPolicy return workload restart policy. Supports both docker-compose and Kubernetes notations.
-func (p *ProjectService) restartPolicy() v1.RestartPolicy {
-	policy := config.RestartPolicyAlways
-	if p.SvcK8sConfig.Workload.RestartPolicy != "" {
-		policy = p.SvcK8sConfig.Workload.RestartPolicy
-	}
-
-	restartPolicy, err := getRestartPolicy(p.Name, policy)
-	if err != nil {
-		log.WarnWithFields(log.Fields{
-			"project-service": p.Name,
-			"restart-policy":  policy,
-		}, "Restart policy is not supported, defaulting to 'Always'")
+// restartPolicy returns workload restart policy
+func (p *ProjectService) restartPolicy() (v1.RestartPolicy, error) {
+	return toV1RestartPolicy(p.SvcK8sConfig.Workload.RestartPolicy)
+}
 
-		return config.RestartPolicyAlways
+// toV1RestartPolicy maps to a case-sensitive v1 restart policy
+func toV1RestartPolicy(rp config.RestartPolicy) (v1.RestartPolicy, error) {
```

Comment:
should this func be moved to `restart.go`? I think this func could just take any previously inferred or config extracted RestartPolicy value and ensure that it returns a valid v1.RestartPolicy. In case the current config value wasn't correct perhaps we could just fall back to default instead of erroring? Though I get why you'd want to error in some cases... What do you think?

Reply:
Funny you should ask this question - because that's where the func started 😄 ...

But I moved it here because its a converter issue. Different converters may need to deal with this value in different ways.

Regarding the error, the likelihood is small but it's a useful guard. Also, useful for a user to rectify any bad config.